### ISSUE TITLE
"Could not find sketch size" error freezes the UI

### ISSUE THREAD

**Issue by JoeManto** [processing/processing#4893]
```
import java.util.Random;
int w = 300;
int h = 300;

void setup(){
     size(w,h);
}
```
Totally freezes the whole processing application. I understand why it freezes but I think there could be some easy error catching in the size class that would prevent the freeze. (possibly a default size) (possible Exception to catch "IllegalArgumentException")

**Why this fix would be helpful** 
One problem for instance if you spend 20 minutes working on a quick sketch and for some reason setup your sizing as provided above. Your gunna be left with a frozen application that can only be closed by force quit, therefor all code is lost and assuming that you didn't save. (which is a very real possibility for only 20 minutes of work)

**Comment by benfry** [comment]
What you describe is already implemented; just not sure why it would be freezing up completely—that's not happening for me. 

<img width="727" alt="screen shot 2017-02-11 at 10 44 06 pm" src="https://cloud.githubusercontent.com/assets/1623101/22859502/b36d8b24-f0ab-11e6-8737-21495e772361.png">

**Comment by JoeManto** [comment]
add a import line above the int var's

**Comment by JoeManto** [comment]
<img width="726" alt="screen shot 2017-02-11 at 10 47 52 pm" src="https://cloud.githubusercontent.com/assets/10517517/22859517/2ccea44e-f0ac-11e6-97f1-a7d16288b7ac.png"> although the error box still frozen

**Comment by benfry** [comment]
Did you close this issue by accident?

**Comment by benfry** [comment]
The problem here is almost certainly an event queue/threading issue. This dialog box isn't properly being launched on the event queue, or it's trying to launch itself on the event queue from a thread that's already there, causing a deadlock.

Probably simple enough to just swap the behavior, but the larger issue is that errors that happen on startup of a sketch (or otherwise inside `setup()`) often cause a spinning beach ball until someone hits Stop. This is related, and should be fixed alongside this problem.

**Comment by depperm** [comment]
I experienced the same thing today. The popup shows but the OK is unclickable and so is the close x.

**Comment by depperm** [comment]
In PdePreprocessor.java line 379 is where this warning is created and then it throws a SketchException. My guess is there is no try...except to handle the exception in JavaBuild.java. There is a handle for that exception but that is down at line 245~374 and line 216 is where initSketchSize function is called which may throw the exception.

**Comment by mictadlo** [comment]
Hi, any update on it? I got the error with processing 3.3.7. Is there a workaround? 

Thank you in advance

Michal

**Comment by Debasishray19** [comment]
Hi, I got the same error with processing 3.3.7
How could we resolve this?

Deb

**Comment by Kripner** [comment]
Got exactly the same error with processing 3.3.7.

**Comment by maf1a** [comment]
damn, i get the same thing. wasted a few hours of work without saving the project

**Comment by shubhamtewari** [comment]
I have processing 3.3.7, on Ubuntu 18.04 LTS and I am not able to re-create this error. It was mentioned above that this problem was occuring in MacOs and also that it might be a deadlock problem. Could this be the reason why I can't recreate it? The dialog box closes like it should. Does anyone get this problem on Linux?

**Comment by IanGolsby** [comment]
I have the issue on Windows with 3.3.6. Is it fixed on 7 yet? That would be good motivation to update lol

**Comment by benfry** [comment]
We've not had anyone help me with finding a fix, so it's not fixed yet. 

If you've lost any code, check your temporary directories since the code is copied there each time you run your sketch (which is why #1896 isn't as easy a fix as one might think).

**Comment by h3rb** [comment]
I have the same problem, Processing 3.4, Windows 10
* cannot click "OK"
* cannot click "X" to close dialog
* cannot hit "ESC" to close dialog
* forced to kill task in Task Manager

**Comment by benfry** [comment]
Locking until we have a solution. It's a known problem, the likely diagnosis is [up thread](https://github.com/processing/processing/issues/4893#issuecomment-297362873). Fixes are welcome, but also need to be tested across all three platforms and with a variety of sketches (different renderers, with and without `draw()` methods, etc).

**Comment by benfry** [comment]
Hopefully fixed for the next release (3.4.1 or 3.5) by https://github.com/processing/processing/pull/5708
